util/remoteconfig.go

```
@@ -0,0 +1,125 @@
+	remoteConfig := &RemoteConfig{}
+
+	//intialize consul
+	if consulHost == "" && consulPort == 0 {
+		remoteConfig.Consul = consulet.Default()
+	} else {
+		remoteConfig.Consul = consulet.New(consulHost, consulPort)
+	}
+
+	//initialize vault
+	if token == "" {
+		vaultClient, err := ocevault.NewEnvAuthClient()
+		if err != nil {
+			return nil, err
+		}
+		remoteConfig.Vault = vaultClient
+	} else {
+		vaultClient, err := ocevault.NewAuthedClient(token)
+		if err != nil {
+			return nil, err
+		}
+		remoteConfig.Vault = vaultClient
+	}
+
+	return remoteConfig, nil
+}
+
+//RemoteConfig is an abstraction for retrieving/setting creds for ocelot
+//currently uses consul + vault
+type RemoteConfig struct {
+	Consul *consulet.Consulet
+	Vault  *ocevault.Ocevault
+}
+
+//GetCredAt will return list of credentials stored at specified path.
+//if hideSecret is set to false, will return password in cleartext
+//key of map is CONFIG_TYPE/ACCTNAME. Ex: bitbucket/mariannefeng
+func (remoteConfig *RemoteConfig) GetCredAt(path string, hideSecret bool) map[string]*models.AdminConfig {
+	creds := map[string]*models.AdminConfig{}
+	for _, v := range remoteConfig.Consul.GetKeyValues(path) {
+		pathKeys := strings.Split(strings.TrimLeft(v.Key, "/" + ConfigPath), "/")
+
+		//cred type | acct name gives us a unique id to track by in the map
+		credType := pathKeys[0]
+		acctName := pathKeys[1]
+		infoType := pathKeys[2]
+
+		mapKey := credType + "/" + acctName
+		foundConfig, ok := creds[mapKey]
+		if !ok {
+			foundConfig = &models.AdminConfig{
+				AcctName: acctName,
+				Type:     credType,
+			}
+
+			if hideSecret {
+				foundConfig.ClientSecret = "*********"
+			} else {
+				passcode, err := remoteConfig.GetPassword(ConfigPath + "/" + credType + "/" + acctName)
+				if err != nil {
+					ocelog.IncludeErrField(err).Error()
+					foundConfig.ClientSecret = "ERROR: COULD NOT RETRIEVE PASSWORD FROM VAULT"
```

i'm thinkin that at this point, we should return an error, because the caller of this is asking for a password, seems like itd be more clear to return an error than the caller doing a string match on ClientSecret to see if it matches that message

---

:+1: I'll make the change to remoteconfig